pkg/splunk/controller/configmap.go

```diff
@@ -75,3 +76,53 @@ func GetConfigMapResourceVersion(client splcommon.ControllerClient, namespacedNa
 	}
 	return configMap.ResourceVersion, nil
 }
+
+// GetMCConfigMap gets the MC ConfigMap resource required for that MC
+func GetMCConfigMap(client splcommon.ControllerClient, namespacedName types.NamespacedName) (*corev1.ConfigMap, error) {
+	var configMap corev1.ConfigMap
+	err := client.Get(context.TODO(), namespacedName, &configMap)
+	if err != nil {
+		//if we don't find mc configmap create and return an empty configmap
+		var current corev1.ConfigMap
+		current.Data = make(map[string]string)
+		current = corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      namespacedName.Name,
+				Namespace: namespacedName.Namespace,
+			},
+			Data: make(map[string]string),
+		}
+		current.ObjectMeta = metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+		}
+
+		err = splutil.CreateResource(client, &current)
+		if err != nil {
+			return nil, err
+		}
+		return &current, nil
+	}
+	return &configMap, nil
+}
+
+// SetConfigMapOwnerRef sets owner references for configMap
+func SetConfigMapOwnerRef(client splcommon.ControllerClient, cr splcommon.MetaObject, namespacedName types.NamespacedName) error {
+	configMap, err := GetConfigMap(client, namespacedName)
+	if err != nil {
+		return err
+	}
+
+	currentOwnerRef := configMap.GetOwnerReferences()
+	// Check if owner ref exists
+	for i := 0; i < len(currentOwnerRef); i++ {
+		if reflect.DeepEqual(currentOwnerRef[i], splcommon.AsOwner(cr, false)) {
+			return nil
+		}
+	}
+
+	// Owner ref doesn't exist, update configmap with owner references
+	configMap.SetOwnerReferences(append(configMap.GetOwnerReferences(), splcommon.AsOwner(cr, false)))
+
+	return splutil.UpdateResource(client, configMap)
```

Comment:
Is the MC Is the only owner for the configMap. Do you see a reason for other CRs also to have ownership of this?

Reply:
my idea is to make that MC only as of the owner of the configmap, don't see an incentive here to make all the Crs owner of that configmap or will it make a difference?